modules/core/src/main/java/org/apache/ignite/internal/util/CollectionUtils.java

```diff
@@ -142,6 +143,41 @@ else if (ts == null || ts.length == 0)
         }
     }
 
+    /**
+     * Create a view for an iterable with conversion of elements.
+     *
+     * @param iterable Iterable.
+     * @param mapper Conversion function.
+     * @param <T1> Type of the elements.
+     * @param <T2> Converted type of the elements.
+     * @return View of iterable.
+     */
+    public static <T1, T2> Iterable<T2> view(
```

Comment:
Why do you need this if Java has streams?

Reply:
In order not to create collections.
WDYT?

Reply:
Replaced with `CollectionUtils#viewReadOnly`